pyomo/solvers/plugins/solvers/mosek_direct.py

```diff
@@ -11,104 +11,92 @@
 import logging
 import re
 import sys
+import itertools
+import operator
+import pyomo.core.base.var
+import pyomo.core.base.constraint
 from pyutilib.misc import Bunch
 from pyutilib.services import TempfileManager
-from pyomo.core.expr.numvalue import is_fixed
-from pyomo.core.expr.numvalue import value
+from pyomo.core.expr.numvalue import (is_fixed,value)
+from pyomo.core.kernel.objective import minimize, maximize
 from pyomo.repn import generate_standard_repn
-from pyomo.solvers.plugins.solvers.direct_solver import \
-    DirectSolver
+from pyomo.core.base.suffix import Suffix
+from pyomo.solvers.plugins.solvers.direct_solver import DirectSolver
 from pyomo.solvers.plugins.solvers.direct_or_persistent_solver import \
-    DirectOrPersistentSolver
-from pyomo.core.kernel.conic import (_ConicBase,
-                                     quadratic,
-                                     rotated_quadratic,
-                                     primal_exponential,
-                                     primal_power,
-                                     dual_exponential,
-                                     dual_power)
-from pyomo.core.kernel.objective import minimize, maximize
-from pyomo.common.collections import ComponentSet, ComponentMap
+        DirectOrPersistentSolver
+from pyomo.common.collections import ComponentMap, ComponentSet
+from pyomo.opt import SolverFactory
+from pyomo.core.kernel.conic import (_ConicBase,quadratic,rotated_quadratic,
+                                     primal_exponential,primal_power,
+                                     dual_exponential,dual_power)
 from pyomo.opt.results.results_ import SolverResults
 from pyomo.opt.results.solution import Solution, SolutionStatus
 from pyomo.opt.results.solver import TerminationCondition, SolverStatus
-from pyomo.opt.base import SolverFactory
-from pyomo.core.base.suffix import Suffix
-import pyomo.core.base.var
-
-
 logger = logging.getLogger('pyomo.solvers')
 
 
 class DegreeError(ValueError):
     pass
 
-
 def _is_numeric(x):
     try:
         float(x)
     except ValueError:
         return False
     return True
 
-
-@SolverFactory.register('mosek', doc='Direct python interface to Mosek')
-class MosekDirect(DirectSolver):
-
-    def __init__(self, **kwds):
-        kwds['type'] = 'mosek'
+@SolverFactory.register('mosek_direct',doc='Direct python interface to MOSEK')
```

Comment:
I had this concern myself and was planning to make a follow-up PR sometime soon with the said "meta" solver. In the meantime, I will admit, I was hoping that this would not be a huge deal-breaker.